### PR TITLE
[chore] Mark issues as stale after 60 days

### LINKED ISSUE #12731: [chore] Track unmaintained issues
With the creation of the [Unmaintained stability level](https://github.com/open-telemetry/opentelemetry-collector#unmaintained) and the ability to [automatically ping code owners when a component label is added](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/12411), we need a way to keep track of issues that do not get any responses from the code owner.  

I think it would be good if we can automatically ping the code owner again in 3 weeks if the issue is still open, the issue is not assigned, and the code owner has not responded.  Then in another 3 weeks (6 weeks total since issue creation), the issue is marked Stale (or something equivalent).

Maintainers would then be able to look for Stale issues and start making judgements if the component should be marked unmaintained.  

The reason I feel this automation is necessary is because this repository gets so many issues, it feels unrealistic to go back through issues manually and check if code owners are responding.

### PR DESCRIPTION
**Pull request by evan-bradley**
**Description:**

This marks issues as stale after 60 days, exempting issues that are labeled with `never stale`. Upon being marked stale, code owners for any corresponding package labels are pinged.

Issues are closed after 180 days (roughly 6 months) of total inactivity (4 months after the `Stale` label is applied), and are given the `closed as inactive` label so they can be easily found later.

**Link to tracking Issue:**
Resolves https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/12731

**Testing:**

I tested this on my personal repository. Here is a screenshot of the bot pinging code owners:

bot pinging code owners on issue

### CODE REVIEW

**Comment by evan-bradley** [comment]
@open-telemetry/collector-contrib-approvers I'd welcome additional input on this approach. I'll also bring this up as a discussion item at tomorrow's SIG meeting.

**Comment by djaglowski** [comment]
I think I'd prefer that the "never-stale" label is applied only by triagers/approvers/maintainers. Otherwise, I think it would be applied inconsistently.

**Comment by evan-bradley** [comment]
@djaglowski That's a fair point. I'll remove it from the workflow.

File: .github/workflows/stale.yaml [hunk 1]
```diff
@@ -13,6 +13,8 @@ jobs:
           stale-pr-message: 'This PR was marked stale due to lack of activity. It will be closed in 14 days.'
           close-pr-message: 'Closed as inactive. Feel free to reopen if this PR is still being worked on.'
           days-before-pr-stale: 14
-          days-before-issue-stale: 730
+          days-before-issue-stale: 60
           days-before-pr-close: 14
-          days-before-issue-close: 30
+          days-before-issue-close: -1
```
[the remark this answers is not in the document]
Comment (evan-bradley) [reply .github/workflows/stale.yaml]: I've set it to roughly 6 months (180 days) following the consensus from the SIG meeting discussion.

**Comment by bogdandrutu** [comment]
We discuss during the SIG meeting:
* Mark stale after 60
* Close after 180 (60 marking + 120)
* Mark issues with a label if possible "close by stale bot"

**Comment by evan-bradley** [comment]
Thanks @bogdandrutu. I've implemented all of these settings for the stale bot. Closed issues are labeled with `closed as inactive`.

**Review by Aneurysm9 — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]

**Review by TylerHelmuth — APPROVED** [review]

**Review by codeboten — COMMENTED** [review]
The changes look good, one question though. Will running `ping-codeowners-for-issue` add a comment triggering the issue to no longer be stale?

**Review by mx-psi — APPROVED** [review]

**Comment by evan-bradley** [comment]
I'm running some tests to confirm the stale action's behavior. I will report back when everything looks good.

**Comment by evan-bradley** [comment]
In regard to whether pinging code owners will remove the stale label, I've decided to have the same workflow mark issues as stale and ping code owners to ensure there is no conflict. The stale action checks when the [stale label was applied](https://github.com/actions/stale/blob/0d6f83007122d952cf222d03c8a3f6460c3a80fe/src/classes/issues-processor.ts#L630) and uses that to determine the last updated date, so a comment by another bot will move this date forward and look like an update to the stale bot. There is technically a 15 second period where we could post a comment after the label has been applied, but I don't want to rely on that and it's not publicly documented behavior anyway. The [behavior to filter out bot comments](https://github.com/actions/stale/blob/0d6f83007122d952cf222d03c8a3f6460c3a80fe/src/classes/issues-processor.ts#L770) that I mentioned previously does apply, but from what I can tell really only matters for comments in that 15 second window.

The "mark issues as stale" workflow will now run on a schedule to check 30 of the least-recently updated issues in the repository which don't have the "Stale" or "never stale" labels. For each retrieved issue not updated for at least 60 days, it will leave a comment that the issue is stale and notify code owners if any are associated with component labels, then mark the issue as stale. The stale bot will then handle removing the stale label if the issue is updated, or closing the issue if another 120 days has elapsed since the stale label was applied. Issues are closed as "not planned" and are given the "closed as stale" label so we can find them later.

I've configured the workflow to start off by marking issues as stale once a week so as not to flood anyone's inbox. Currently about 500 issues in the contrib repository would be considered stale. So at a rate of 30 issues/week, the workflow should finish processing currently-stale issues over the next 20 weeks or so when also considering issues that become stale in that period.

**Comment by evan-bradley** [comment]
I've tested this on my personal fork and haven't encountered any issues. I've included a screenshot of what it looks like below. Note that I set the "stale" and "close" periods to 0 days so they take effect as soon as the workflows run.

stale issue test

**Review by djaglowski — APPROVED** [review]

**Review by TylerHelmuth — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**